### ISSUE TITLE
Cannot reproduce the robustness results in Tensorflow

### ISSUE THREAD

**Issue by GuanlinLee** [cuge1995/PointCutMix#5]
I use the PointCutMix-k augmentation method provided in this repo to try a DGCNN in Tensorflow. And attack the model with Saliency Map Attack(20iters, 5 points dropped in each iter). However, the accuracy under attack is only 80.97%, which is far lower than results in your paper.

**Comment by cuge1995** [comment]
hey, thanks for your attention, can you opensource your code?

**Comment by GuanlinLee** [comment]
The code I use is from https://github.com/WangYueFt/dgcnn/tree/master/tensorflow. I use the data augmentation part from your PointCutMix-k. I transform the numpy data into torch.tensor to obtain the mixed point cloud, and then I transform them into numpy to feed them into the tensorflow model. You can reproduce the result from this repo and attack the model with the Saliency Map Attack, which is from https://github.com/tianzheng4/PointCloud-Saliency-Maps.

**Comment by cuge1995** [comment]
What's the results did your point out? In table 4 or 5?

**Comment by GuanlinLee** [comment]
Table 5, of course. DGCNN can achieve 91.05% under Drop 100 attack. However, I only obtain 80.97%, when using Tensorflow. By the way, the DGCNN I use is from https://github.com/tianzheng4/PointCloud-Saliency-Maps.

**Comment by GuanlinLee** [comment]
Sorry, I find the attack in your paper is Black-Box. I test the model under White-Box. Can you provide more accuracy under White-Box attack? Because the Black-Box is not convincing enough to show the robustness, and there are no gradient masking in your model, the White-Box attacks can be easy to do.

**Comment by GuanlinLee** [comment]
Actually, if without using the PointCutMix-k to do data augmentation, the model can achieve 80%~81% accuracy under White-Box Saliency Map Attack as well.

**Comment by cuge1995** [comment]
> Sorry, I find the attack in your paper is Black-Box. I test the model under White-Box. Can you provide more accuracy under White-Box attack? Because the Black-Box is not convincing enough to show the robustness, and there are no gradient masking in your model, the White-Box attacks can be easy to do.

I think the results in Table 4 are what you want.

**Comment by GuanlinLee** [comment]
Yes, accuracy in Table 4 is 85.86%. However, I can only achieve 80.97%, which is closer to the accuracy of the model without any data argumentation. The baseline, which is without using PointCutMix-k, in Tensorflow of DGCNN is 80.64%. And the baseline in Tensorflow of PointNet is 57.39%. I think there are some mistakes in your baselines.

**Comment by cuge1995** [comment]
> Yes, accuracy in Table 4 is 85.86%. However, I can only achieve 80.97% when I only drop 100 points, which is closer to the accuracy of the model without any data argumentation. The baseline, which is without using PointCutMix-k and dropping 100 points, in Tensorflow of DGCNN is 80.64%. And the baseline in Tensorflow of PointNet is 57.39%. I think there may be some mistakes in your baselines or robust models. Or could you try to test on Tensorflow?

Please note that our results in Table 4 are drop 200 attack, and please make sure all the training scheme, standard data augmentation, learning rate, the checkpoint is the best test result's model are the same.

The baseline in Table 4 was adopted from IF-Defense, and actually, our baseline is lower than that, we will update our baseline in the future version.

I am not familiar with TF, but I think the difference shouldn't be that big, please check your code carefully.

**Comment by cuge1995** [comment]
> Actually, if without using the PointCutMix-k to do data augmentation, the model can achieve 80%~81% accuracy under White-Box Saliency Map Attack as well.

I think the key difference is the drop attack of the IF-Defense baseline and the original [drop attack](https://github.com/tianzheng4/PointCloud-Saliency-Maps).

**Comment by GuanlinLee** [comment]
Could you release the attack code as well for better evaluation on the robustness? Because I find officially released attacks are based on TF.

**Comment by cuge1995** [comment]
hey, we used the [IF-Defense baseline](https://github.com/Wuziyi616/IF-Defense/tree/main/baselines) to produce our results, I strongly recommend it.

**Comment by GuanlinLee** [comment]
I have spent a day to implement all parts of your methods. Here are some updates. I can achieve 81% accuracy under drop-200, which I think is acceptable, for there are still some differences between Tensorflow and Pytorch. However, when I try to do some ablation studies on every part of the code, I find that the label smoothing and data argumentation form PointcloudScaleAndTranslate improve the robustness of models more than PointCutMix-k. You could explore more about this. I think the statement that PointCutMix can enhance the robustness is kind of exaggerating. Hope this is useful for your work.

**Comment by cuge1995** [comment]
Huge thanks for your great work.

But I was wondering how did you validate the PointcloudScaleAndTranslate improve the robustness of models more than PointCutMix-k? Because we train models with PointCutMix-k and PointcloudScaleAndTranslate augmentation at the same time.

Anyway, thanks for the contribution, and I hope those findings may useful in your future research.

**Comment by cuge1995** [comment]
If you are willing to open source the reproduced TF code, I will be very grateful and recommend it in my repo.

**Comment by GuanlinLee** [comment]
> Huge thanks for your great work.
> 
> But I was wondering how did you validate the PointcloudScaleAndTranslate improve the robustness of models more than PointCutMix-k? Because we train models with PointCutMix-k and PointcloudScaleAndTranslate augmentation at the same time.
> 
> Anyway, thanks for the contribution, and I hope those findings may useful in your future research.

Oh, I just replace the PointcloudScaleAndTranslate with a simple jitter, shifting and rotating translation. I need to do more explorations on my reproduced code before releasing it.

**Comment by GuanlinLee** [comment]
I find a bug in my Tensorflow code. When I use data augmentation like rotating and shifting, two point clouds may be not aligned. It can decrease the robustness when under attacks. So I see you only use the scale transformation before mix up them. After more experiments, the robustness brought by the PointCutMix is significant. There are no more questions from me. Thanks.

And for the Tensorflow code, I think I would release them later.

**Comment by cuge1995** [comment]
> I find a bug in my Tensorflow code. When I use data augmentation like rotating and shifting, two point clouds may be not aligned. It can decrease the robustness when under attacks. So I see you only use the scale transformation before mix up them. After more experiments, the robustness brought by the PointCutMix is significant. There are no more questions from me. Thanks.
> 
> And for the Tensorflow code, I think I would release them later.

If you are interested in adversarial robustness, I recommend you to look at our newly released paper [IT-Defense](https://github.com/cuge1995/IT-Defense).
